src/cdomains/intDomain.ml

```diff
@@ -1750,8 +1755,57 @@ module Enums : S with type int_t = BigInt.t = struct
   let is_excl_list = BatOption.is_some % to_excl_list
   let starting     ikind x = top_of ikind
   let ending       ikind x = top_of ikind
-  let maximal = function Inc xs when not (ISet.is_empty xs) -> Some (ISet.max_elt xs) | _ -> None
-  let minimal = function Inc xs when not (ISet.is_empty xs) -> Some (ISet.min_elt xs) | _ -> None
+
+  let lognot ik x =
+    if is_bot x
+    then x
+    else
+      match to_bool x with
+       | Some b -> of_bool ik (not b)
+       | None -> top_bool
+
+  let logand = lift2 I.logand
+  let logor  = lift2 I.logor
+  let maximal = function
+    | Inc xs when not (ISet.is_empty xs) -> Some (ISet.max_elt xs)
+    | Exc (_,r) -> Size.max_from_bit_range (R.maximal r)
```

Comment:
Is it sound to ignore the excluded elements here?

Reply:
The semantics of the `maximal` function should be to return an upper bound of the the concrete values, right? So this is sound, but possibly imprecise, because it might not be the smallest upper bound.

Reply:
I added some check in ac4d7d4817ce498ccb994a009b26811c0b3f5977 to make the values more precise in case the maximum given by the range is in the excluded set. 
I assume that this will happen pretty rarely and I am not sure whether we should do this?

Reply:
One case where this might be interesting is if it allows excluding an overflow of sorts. Not sure if this realistically happens much